File: tests/fixtures/springboot-comprehensive/src/main/java/com/example/users/UserController.java
package com.example.users;

import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/users")
public class UserController {

    @GetMapping
    public List<User> list() { return List.of(); }

    @PostMapping
    public User create(@RequestBody @Valid User body) { return body; }

    @GetMapping("/{id}")
    public User show(@PathVariable String id) { return new User(); }

    @PutMapping("/{id}")
    public User update(@PathVariable String id, @RequestBody @Valid User body) { return body; }

    @DeleteMapping("/{id}")
    public void delete(@PathVariable String id) {}
}

File: tests/smoke-fixtures/nestjs-mini/src/user.controller.ts
import { Controller, Get, Post, Put, Delete, Body, Param } from "@nestjs/common";

@Controller("users")
export class UserController {
  @Get()       list()                          { return []; }
  @Post()      create(@Body() body: unknown)   { return body; }
  @Get(":id")  show(@Param("id") id: string)   { return { id }; }
  @Put(":id")  update(@Param("id") id: string, @Body() body: unknown) { return body; }
  @Delete(":id") remove(@Param("id") id: string) { return {}; }
}
